### ISSUE TITLE
[v4] Parsing issue around build.esbuild.exclude and build.esbuild.external

### ISSUE THREAD

**Issue by kdybicz** [serverless/serverless#12617]
### Issue description

Based on the documentation found [here](https://www.serverless.com/framework/docs/providers/aws/guide/building) I was trying to exclude some dependencies from my build using such config:
```
build:
  esbuild:
    bundle: true
    minify: false
    sourcemap: false
    exclude:
      - @aws-sdk/*
```
When running `npx sls package --stage staging` I got:
```
✖ Could not read the config file at path "/Users/kamil/work/Phoenix/serverless.yml" due to this error: Cannot parse "serverless.yml": bad indentation of a sequence entry in "/Users/kamil/work/Phoenix/serverless.yml" (91:9)

 88 |     minify: false
 89 |     sourcemap: false
 90 |     exclude:
 91 |       - @aws-sdk/*
--------------^
 92 | 
 93 | # https://www.serverless.com/framework/ ...
```

The same issue applies to:
```
    external:
      - @aws-sdk/client-s3
```

And it's being fixed by using `"`:
```
    exclude:
      - "@aws-sdk/*"
```

### Context

# Service Overview
- Serverless Framework Version: 4.1.5
- Service Path: /Users/kamil/work/Phoenix
- Service Config File: serverless.yml
- Service Name: phoenix
- Service Runtime: nodejs20.x
- Service Stage: staging
- Service Region: eu-west-1
- Command: package --stage staging

**Comment by kdybicz** [comment]
This might be just a documentation issue though

**Comment by eahefnawy** [comment]
Thanks @kdybicz ... This was fixed in https://github.com/serverless/serverless/pull/12646 and the docs will be updated shortly after merging

**Comment by Mmarzex** [comment]
This has been released with [4.1.7](https://github.com/serverless/serverless/releases/tag/v4.1.7)
